src/components/buttons/index.js

```diff
@@ -0,0 +1,26 @@
+import React from "react"
+import PropTypes from "prop-types"
+import { useDispatch } from "react-redux"
+import { isAdding } from "redux/actions"
+import { useSelector } from "react-redux"
+
+
+function Button({ text }) {
+  const dispatch = useDispatch()
+  const selectIsAdding = (state) => state.todos.isAdding
+  const toggleIsAdding = useSelector(selectIsAdding)
+
+  function handleClick() {
+    if (text === "Create") {
+      dispatch(isAdding(!toggleIsAdding))
+    }
+  }
```

Comment:
Can we lift this checking at top of its parent component. As we are reusing this `Button` most of the cases this condition checking are not needed.